### ISSUE TITLE
How to access localhost?

### ISSUE THREAD

**Issue by kikoShen** [regisb/tutor#213]
I'm new to tutor and looking for a way to run open edx on my Mac withou a virtual box or Ubuntu. Here's 3 questions in total:
1. I use the latest version and followed the instructions ,start the services successfully by running `tutor local quickstart`. But I can't access http://localhost either http://localhost:8000 . Got a normal error : 403 forbidden You don't have permission to access / on this server.
Here's my config.yaml:
`CMS_HOST:studio.localhost` 
`LMS_HOST: localhost`
`LANGUAGE_CODE: zh-cn`
Note : I can access  by running  `tutor dev runserver lms` .

2. In the [docs>Using Tutor for Open edX development](https://docs.tutor.overhang.io/dev.html),I notice it said :

> You should setup real host names for the LMS and the CMS (i.e: not “localhost”). 

The problem is what's my real host names？
Note:
I run `$hostname` and got `kiko.local`
and tried to configure the `lms_host= kiko.local`   and ` cms_host = studio.kiko.local` . It didn't work!

3. while I run decompose command `tutor local start` ,it seems like the console won't give me a message to tell me that it was success or end. Is that true ? There always something printed.

Umm, I know it's not  a good behavior to write 3 questions in only issue. but I guess these questions are simple so I asked it by the way. I'll take care next time!

**Comment by regisb** [comment]
Hi @kikoShen! Indeed, the three questions are somewhat related. I don't believe it's a bug, so maybe it would be a better idea to move this conversation to http://discuss.overhang.io/
Anyway, let's try to solve your problems here.

1. You must define realistig hostnames that will identify your platform. It doesn't matter if they are not the "real" hostnames of your local computer if you are only interested in development. Try running:

    tutor config save --set LMS_HOST=myopenedx.com --set CMS_HOST=studio.myopenedx.com -y
    tutor local stop
    tutor local start

You should then be able to access the lms at http://localhost and http://studio.localhost.

2. > The problem is what's my real host names？
Basically, anything that does not end in "localhost" :)

3. `tutor local start` does not run in daemon mode. To run in daemon mode run `tutor local start -d` (see https://docs.tutor.overhang.io/local.html#running-open-edx)

**Comment by regisb** [comment]
I closed this but feel free to comment.

**Comment by kikoShen** [comment]
Thanks for your reply  :)
Umm，I tried running all 3 steps you listed  on question 1. but there's still 403 error link to http://localhost. As for myopenedx.com ,can't access too.


>Could you give me a more specific explanation for this?
   You must define realistig hostnames that will identify your platform.

I guess it must be something  I miss out. And the point I can't get into is that what IP address `myopenedx.com `link to ? I don't have an registered domain name for a site now. Is that mean I should buy one?

BTW, I mapped the apache to port 444 ,nginx to port 81. but I don't think  it relate to this problem because I can start the service.

**Comment by kikoShen** [comment]
OMG. I failed in this step:
> sudo ln -s $(tutor config printroot)/env/local/proxy/nginx/openedx.conf /etc/nginx/sites-enabled/
ln: /etc/nginx/sites-enabled/: No such file or directory

Unfortunately ,even don't have /etc/nginx/ directory. That means I should download nginx first ?:(

**Comment by regisb** [comment]
> That means I should download nginx first ?:(
No! Let's backtrack a bit. Why did you remap the nginx ports?

**Comment by kikoShen** [comment]
I've always thought Apache and Nginx both required. At first time I run` tutor local start`, there's an error ...anyway, I don't remember... but appear to be the problem relate to [https://docs.tutor.overhang.io/local.html#running-open-edx-behind-a-web-proxy](https://docs.tutor.overhang.io/local.html#running-open-edx-behind-a-web-proxy).

Now I change port back to 80 and 443 ,it give me error :

> ERROR: for nginx  Cannot start service nginx: driver failed programming external connectivity on endpoint tutor_local_nginx_1 (cd1ed59b42987a53eff0f118214e75c1226c66aa95182b58a4c2aedb78c319a8): Error starting userland proxy: Bind for 0.0.0.0:80: unexpected error (Failure EADDRINUSE)
ERROR: Encountered errors while bringing up the project.
Error: Command failed with status 1: docker-compose -f /Users/kiko/Library/Application Support/tutor/env/local/docker-compose.yml --project-name tutor_local up --remove-orphans -d

Now I just realized that  Apache and Nginx just do the same thing maybe... So if i have apache already , I don't need  nginx ?

**Comment by regisb** [comment]
> Now I just realized that Apache and Nginx just do the same thing maybe... So if i have apache already , I don't need nginx ?

You don't need either. Did you get this impression from the tutor documentation?
If you installed apache just for tutor, you can safely uninstall it.

**Comment by kikoShen** [comment]
Ummm, I installed apache for development Java and php before. 
To keep apache,  I just stop apache service ,and start tutor.  At first I got error:504 GATE WAY TIMEOUT. but a few minutes later, I can access http://localhost. :)
Thanks a lot! 👍
